python/lsst/daf/butler/formatters/yamlFormatter.py

```diff
@@ -0,0 +1,97 @@
+# This file is part of daf_butler.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (http://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import builtins
+import yaml
+
+from lsst.daf.butler.formatters.fileFormatter import FileFormatter
+
+
+class YamlFormatter(FileFormatter):
+    """Interface for reading and writing Python objects to and from YAML files.
+    """
+    extension = ".yaml"
+
+    def _readFile(self, path, pytype=None):
+        """Read a file from the path in YAML format.
+
+        Parameters
+        ----------
+        path : `str`
+            Path to use to open YAML format file.
+
+        Returns
+        -------
+        data : `object`
+            Either data as Python object read from YAML file, or None
+            if the file could not be opened.
```

Comment:
I'm a bit surprised that we don't raise an exception when we can't find the file.  Is this the pattern for all formatters?

Reply:
It made `FileFormatter.read()` a little less convoluted and finding the file was the only thing that mattered. I think this code will have to be rejiggered when we start using templates for file names so I'll have another look at that time. @pschella didn't like including the component name in the filename (I think he preferred it to be under the control of the template) so this will change.